public/docs/_examples/cb-item-template/ts/app/simple-select.component.ts

```diff
@@ -0,0 +1,85 @@
+// #docplaster
+// #docregion
+import { Component } from '@angular/core';
+import { ContentChild } from '@angular/core';
+import { EventEmitter } from '@angular/core';
+import { Input } from '@angular/core';
+import { Output } from '@angular/core';
+import { TemplateRef } from '@angular/core';
+
+interface ItemContext {
+  item: any;
+  index: number;
+}
+
+// #docregion component
+// #docregion metadata
+@Component({
+  selector: 'simple-select',
+  template:
+  `
+    <button (click)='toggleItems()' class='select-root'>
+      
+      <template 
+        [ngTemplateOutlet]='itemTemplateRef'
+        [ngOutletContext]='{ item: value, index: -1 }'>
+      </template>
+
+    </button>
+  
+    <ul *ngIf='isShowingItems' class='select-items'>
+      <li *ngFor='let item of items ; let index = index ;' (click)='selectItem( item )'>
+        
+        <template 
+          [ngTemplateOutlet]='itemTemplateRef'
+          [ngOutletContext]='{ item: item, index: index }'>
+        </template>
+
+      </li>
+    </ul>
+  `
+})
+// #enddocregion metadata
+export class SimpleSelectComponent {
+
+  public isShowingItems: boolean;
+  @Input() public items: any[];
+  public itemTemplateRef: TemplateRef<ItemContext>;
+  @Input() public value: any;
+  @Output() public valueChange: EventEmitter<any>;
+
+  constructor() {
+    this.isShowingItems = false;
+    this.items = [];
+    this.itemTemplateRef = null;
+    this.value = null;
+    this.valueChange = new EventEmitter();
+  }
```

Comment:
I've sort of just developed that style that the property definition says "what it is" and the constructor says "what value it has." Mostly, I'm just making it up as I go along. That said, outside of TypeScript, I don't believe that you can actually initialize class properties as part of the prototype. So, putting the values in the constructor may keep the code truer to the ES6 spec? But, like you, I don't feel very strong. I'll do what ever you think keeps it most consistent.